dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/InterpretationController.java

```diff
@@ -59,60 +72,117 @@
  */
 @Controller
 @RequestMapping( value = InterpretationSchemaDescriptor.API_ENDPOINT )
 public class InterpretationController
-    extends AbstractCrudController<Interpretation>
+    extends
+    AbstractCrudController<Interpretation>
 {
     @Autowired
     private InterpretationService interpretationService;
 
     @Autowired
     private IdentifiableObjectManager idObjectManager;
 
+    @Override
+    @SuppressWarnings( "unchecked" )
+    protected List<Interpretation> getEntityList( WebMetadata metadata, WebOptions options, List<String> filters,
+        List<Order> orders )
+        throws QueryParserException
+    {
+
+        // If mentions:in:[username] in filters -> Remove from filters and add as disjunction
+        List<String> mentions = new ArrayList<String>();
+        ListIterator<String> filterIterator = filters.listIterator();
+        while ( filterIterator.hasNext() )
+        {
+            String[] filterSplit = filterIterator.next().split( ":" );
+            if ( filterSplit[1].equals( "in" ) && filterSplit[0].equals( "mentions" ) )
+            {
+                mentions.add( filterSplit[2] );
+                filterIterator.remove();
+            }
+        }
+
+        Query query = queryService.getQueryFromUrl( getEntityClass(), filters, orders, options.getRootJunction() );
+        query.setDefaultOrder();
+        query.setDefaults( Defaults.valueOf( options.get( "defaults", DEFAULTS ) ) );
+
+        // If mentions:in:[username] in filters -> Add as disjunction including interpretation mentions and comments mentions
+        for ( String m : mentions )
+        {
+            Disjunction disjunction = query.disjunction();
+            String[] split = m.substring( 1, m.length() - 1 ).split( "," );
+            List<String> items = Lists.newArrayList( split );
+            disjunction.add( Restrictions.in( "mentions.username", items ) );
+            disjunction.add( Restrictions.in( "comments.mentions.username", items ) );
+            query.add( disjunction );
+        }
+
+        List<Interpretation> entityList;
+        if ( options.getOptions().containsKey( "query" ) )
+        {
+            entityList = Lists.newArrayList( manager.filter( getEntityClass(), options.getOptions().get( "query" ) ) );
+        }
+        else
+        {
+            entityList = (List<Interpretation>) queryService.query( query );
+        }
+        return entityList;
+
+    }
+
     // -------------------------------------------------------------------------
     // Intepretation create
     // -------------------------------------------------------------------------
 
-    @RequestMapping( value = "/reportTable/{uid}", method = RequestMethod.POST, consumes = { "text/html", "text/plain" } )
-    public void writeReportTableInterpretation(
-        @PathVariable( "uid" ) String reportTableUid,
+    @RequestMapping( value = "/reportTable/{uid}", method = RequestMethod.POST, consumes = { "text/html",
```

Comment:
Minor: apparently Eclipse re-formated a lot of (unchanged) lines in this file so it's harder what's being modified.

Reply:
I have reverted those changes. I am using the Dhis2 Code Style for Eclipse but it looks like this file was never formatted properly. It should be easier for you know to check the changes...